src/node-file-trace.ts

```diff
@@ -294,13 +294,34 @@ export class Job {
   }
 
   async emitDependency (path: string, parent?: string, filesToEmit?: FilesToEmit) {
-    if (this.processed.has(path)) return;
-    this.processed.add(path);
-
+    if (this.processed.has(path) && this.emitDependencyCache.has(path)) {
+      const cacheItem = this.emitDependencyCache.get(path)
```

Comment:
Let's move 298 above 297 so that we can get and check for existence in the same line.

(not sure if there is a scenario where we remove from emitDependencyCache)